e2e/e2e_tests.py

```diff
@@ -0,0 +1,151 @@
+import os
+from time import sleep
+import unittest
+import subprocess
+
+try:
+    import requests
+    from bs4 import BeautifulSoup
+except ImportError:
+    requirements_file = os.path.join(os.path.dirname(__file__), 'requirements.txt')
+    subprocess.check_call("pip install -r {}".format(requirements_file).split(' '))
```

Comment:
I agree that it this is ugly, but out of practicality I think it makes sense:

You need to install a virtualenv with the dependencies and remember to activate them every time you run those tests. Since we're mostly using docker for development, the chances are quite slim that it is activated already. Furthermore, using the ./test.sh, if you had forgotten to activate it beforehand, you'd need to rerun the entire test-suite again (the long running nature is a mere symptom and shouldn't be used as an argument - but it is a symptom we currently can't get rid of).

Should be possible using:
```
import pip
pip.main(['install', package])
```
Will change it.